sync/src/relayer/transactions_process.rs

```diff
@@ -0,0 +1,176 @@
+use crate::relayer::compact_block::RelayTransactions;
+use crate::relayer::Relayer;
+use ckb_logger::debug_target;
+use ckb_network::{CKBProtocolContext, PeerIndex};
+use ckb_protocol::RelayTransactions as FbsRelayTransactions;
+use ckb_store::ChainStore;
+use failure::Error as FailureError;
+use fnv::FnvHashSet;
+use futures::{self, future::FutureResult, lazy};
+use std::convert::TryInto;
+use std::sync::Arc;
+use std::time::Duration;
+
+const DEFAULT_BAN_TIME: Duration = Duration::from_secs(3600 * 24 * 3);
+
+pub struct TransactionsProcess<'a, CS> {
+    message: &'a FbsRelayTransactions<'a>,
+    relayer: &'a Relayer<CS>,
+    nc: Arc<dyn CKBProtocolContext + Sync>,
+    peer: PeerIndex,
+}
+
+impl<'a, CS: ChainStore + Sync + 'static> TransactionsProcess<'a, CS> {
+    pub fn new(
+        message: &'a FbsRelayTransactions,
+        relayer: &'a Relayer<CS>,
+        nc: Arc<CKBProtocolContext + Sync>,
+        peer: PeerIndex,
+    ) -> Self {
+        TransactionsProcess {
+            message,
+            relayer,
+            nc,
+            peer,
+        }
+    }
+
+    pub fn execute(self) -> Result<(), FailureError> {
+        let relay_txes: RelayTransactions = (*self.message).try_into()?;
+
+        let mut txes = Vec::with_capacity(relay_txes.transactions.len());
+
+        {
+            let tx_filter = self.relayer.shared().tx_filter();
+            for relay_tx in relay_txes.transactions {
```

Comment:
Please use `filter_map` and `collect`

Reply:
After changing to `fliter_map`, Clippy suggests changing to `filter`.